src/install/lockfile.zig

```diff
@@ -1225,7 +1233,11 @@ pub const Printer = struct {
 
                         try writer.writeAll(name);
                         try writer.writeByte('@');
-                        try writer.writeAll(version_name);
+                        if (version_name.len == 0) {
+                            try std.fmt.format(writer, "^{any}", .{version_formatter});
```

Comment:
They might change `any` to mean "use the builtin std.fmt.format" in https://github.com/ziglang/zig/pull/15093

so maybe this should be `{}` 

```suggestion
                            try std.fmt.format(writer, "^{}", .{version_formatter});
```